python/ray/serve/_private/version.py

```diff
@@ -18,14 +29,16 @@ def __init__(self, code_version: Optional[str], user_config: Optional[Any] = Non
             self.unversioned = False
             self.code_version = code_version
 
-        self.user_config = user_config
-        # TODO(simon): make this xlang compatible
-        if isinstance(user_config, bytes):
-            serialized_user_config = user_config
-        else:
-            serialized_user_config = str.encode(json.dumps(user_config, sort_keys=True))
-        self.user_config_hash = crc32(serialized_user_config)
-        self._hash = crc32(serialized_user_config + self.code_version.encode("utf-8"))
+        self.deployment_config: DeploymentConfig = deployment_config
```

Comment:
Please add a comment for this field indicating the behavior (its options may be mutated over time so all logic that uses them should access this field directly)

Reply:
Added!